manifests/init.pp

```diff
@@ -46,22 +46,12 @@
   Hash    $files             = {},
   Boolean $files_hiera_merge = false,
 ) {
-  # variable handling
-  if $files_hiera_merge =~ Boolean {
-    $files_hiera_merge_bool = $files_hiera_merge
-  } else {
-    $files_hiera_merge_bool = str2bool($files_hiera_merge)
+  case $files_hiera_merge {
```

Comment:
Maybe for a later pull request, but should we remove module Hiera merge support altogether as we have done previously and rely on native Hiera merge support?

Might require a new major release though(?)